common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java

```diff
@@ -96,19 +106,106 @@ public final void setThreadLocalMap(InternalThreadLocalMap threadLocalMap) {
 
     /**
      * Returns {@code true} if {@link FastThreadLocal#removeAll()} will be called once {@link #run()} completes.
+     *
+     * @deprecated Use {@link FastThreadLocalThread#currentThreadWillCleanupFastThreadLocals()} instead
      */
+    @Deprecated
     public boolean willCleanupFastThreadLocals() {
         return cleanupFastThreadLocals;
     }
 
     /**
      * Returns {@code true} if {@link FastThreadLocal#removeAll()} will be called once {@link Thread#run()} completes.
+     *
+     * @deprecated Use {@link FastThreadLocalThread#currentThreadWillCleanupFastThreadLocals()} instead
      */
+    @Deprecated
     public static boolean willCleanupFastThreadLocals(Thread thread) {
         return thread instanceof FastThreadLocalThread &&
                 ((FastThreadLocalThread) thread).willCleanupFastThreadLocals();
     }
 
+    /**
+     * Returns {@code true} if {@link FastThreadLocal#removeAll()} will be called once {@link Thread#run()} completes.
+     */
+    public static boolean currentThreadWillCleanupFastThreadLocals() {
+        // intentionally doesn't accept a thread parameter to work with ScopedValue in the future
+        Thread currentThread = currentThread();
+        if (currentThread instanceof FastThreadLocalThread) {
+            return ((FastThreadLocalThread) currentThread).willCleanupFastThreadLocals();
+        }
+        return isFastThreadLocalVirtualThread();
+    }
+
+    /**
+     * Returns {@code true} if this thread supports {@link FastThreadLocal}.
+     */
+    public static boolean currentThreadHasFastThreadLocal() {
+        // intentionally doesn't accept a thread parameter to work with ScopedValue in the future
+        return currentThread() instanceof FastThreadLocalThread || isFastThreadLocalVirtualThread();
+    }
+
+    private static boolean isFastThreadLocalVirtualThread() {
+        long[] arr = fallbackThreads.get();
+        if (arr == null) {
+            return false;
+        }
+        return Arrays.binarySearch(arr, Thread.currentThread().getId()) >= 0;
```

Comment:
beware using `getid` 

see 

https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#getId--

> When a thread is terminated, this thread ID may be reused.

vs https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/lang/Thread.html#threadId()

> The thread ID is unique and remains unchanged during its lifetime.

Reply:
TIL, but it delegates to threadId in newer versions anyway.

Reply:
I still think we should change it just to make it more explicit

Reply:
@normanmaurer threadId is 19+ only unfortunately

Reply:
@yawkat  Is it a compromise solution that I hope can bring you some inspiration 
https://github.com/m1ngyuan/netty/commit/3735b18d9261a52ac26c984a4ffe7eb6c0432353

Reply:
I don't think this is worth using PlatformDependent for. Even with the old `getId` semantics, the code still works fine, since it's guaranteed that the thread id won't stay in the `fallbackThreads` array after the thread is destroyed. And using PlatformDependent doesn't actually fix anything: On those platforms where `threadId` is available, `getId` already delegates to it, while on older versions `getId` is the only thing we can use.

Reply:
alright... I think it is fine as it is